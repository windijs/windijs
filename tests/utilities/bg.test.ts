import { createUtility, backgroundAttachment, backgroundAttachmentConfig, backgroundColor, backgroundGeneric, backgroundClip, backgroundClipConfig, colors, styleProperty, px, rem, calc, hover, focus, notChecked, evenOfType, dark, sm, internalGet, rgb } from "../../src";

test("Background Color", () => {
  const bg = createUtility("bg")
    .use(backgroundColor(colors))
    .init();

  expect(bg.current.css).toMatchSnapshot();
  expect(bg.white.css).toMatchSnapshot();
  expect(bg.blue[700].css).toMatchSnapshot();
  expect(bg.red[500].css).toMatchSnapshot();
});

test("Background Color With Opacity", () => {
  const bg = createUtility("bg")
    .use(backgroundColor(colors))
    .init();
  expect(bg.black.opacity(50).css).toMatchSnapshot();
  expect(bg.amber[100].opacity(50).css).toMatchSnapshot();
});

test("Background Color With Different Opacity Name", () => {
  const bg = createUtility("bg")
    .use(backgroundColor(colors, true, "--my-opacity"))
    .init();
  expect(bg.black.opacity(50).css).toMatchSnapshot();
  expect(bg.amber[100].opacity(50).css).toMatchSnapshot();
});

test("Background Color Without Opacity", () => {
  const bg = createUtility("bg")
    .use(backgroundColor(colors, false))
    .init();
  expect(bg.black.css).toMatchSnapshot();
  expect(bg.amber[100].css).toMatchSnapshot();
});

test("Background Color With Different Color Value Type", () => {
  const customColors = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    dark: "#1c1c1e",
    rgb: "rgb(22, 13, 14)",
    rgba: "rgba(22, 13, 14, 0.5)",
    hsl: "hsl(360, 100%, 50%)",
    hsla: "hsla(360, 100%, 50%, 0.3)",
    hwb: "hwb(194 0% 0%)",
    hwba: "hwb(194 0% 0% / .5)",
  };
  const bg = createUtility("bg")
    .use(backgroundColor(customColors))
    .init();

  expect(bg.inherit.css).toMatchSnapshot("inherit");
  expect(bg.current.css).toMatchSnapshot("current");
  expect(bg.transparent.css).toMatchSnapshot("transparent");
  expect(bg.black.css).toMatchSnapshot("black");
  expect(bg.black.opacity(50).css).toMatchSnapshot("black-opacity");
  expect(bg.dark.css).toMatchSnapshot("dark");
  expect(bg.dark.opacity(50).css).toMatchSnapshot("dark-opacity");
  expect(bg.rgb.css).toMatchSnapshot("rgb");
  expect(bg.rgb.opacity(50).css).toMatchSnapshot("rgb-opacity");
  expect(bg.rgba.css).toMatchSnapshot("rgba");
  expect(bg.rgba.opacity(50).css).toMatchSnapshot("rgba-opacity");
  expect(bg.hsl.css).toMatchSnapshot("hsl");
  expect(bg.hsl.opacity(50).css).toMatchSnapshot("hsl-opacity");
  expect(bg.hsla.css).toMatchSnapshot("hsla");
  expect(bg.hsla.opacity(50).css).toMatchSnapshot("hsla-opacity");
  expect(bg.hwb.css).toMatchSnapshot("hwb");
  expect(bg.hwb.opacity(50).css).toMatchSnapshot("hwb-opacity");
  expect(bg.hwba.css).toMatchSnapshot("hwba");
  expect(bg.hwba.opacity(50).css).toMatchSnapshot("hwba-opacity");
});

test("Background Color Without Opacity With Different Color Value Type", () => {
  const customColors = {
    inherit: "inherit",
    current: "currentColor",
    transparent: "transparent",
    black: "#000",
    dark: "#1c1c1e",
    rgb: "rgb(22, 13, 14)",
    rgba: "rgba(22, 13, 14, 0.5)",
    hsl: "hsl(360, 100%, 50%)",
    hsla: "hsla(360, 100%, 50%, 0.3)",
    hwb: "hwb(194 0% 0%)",
    hwba: "hwb(194 0% 0% / .5)",
  };
  const bg = createUtility("bg")
    .use(backgroundColor(customColors, false))
    .init();

  expect(bg.inherit.css).toMatchSnapshot("inherit");
  expect(bg.current.css).toMatchSnapshot("current");
  expect(bg.transparent.css).toMatchSnapshot("transparent");
  expect(bg.black.css).toMatchSnapshot("black");
  expect(bg.dark.css).toMatchSnapshot("dark");
  expect(bg.rgb.css).toMatchSnapshot("rgb");
  expect(bg.rgba.css).toMatchSnapshot("rgba");
  expect(bg.hsl.css).toMatchSnapshot("hsl");
  expect(bg.hsla.css).toMatchSnapshot("hsla");
  expect(bg.hwb.css).toMatchSnapshot("hwb");
  expect(bg.hwba.css).toMatchSnapshot("hwba");
});

test("Background Attachment", () => {
  const bg = createUtility("bg")
    .use(backgroundAttachment(backgroundAttachmentConfig))
    .init();

  expect(bg.fixed.css).toMatchSnapshot();
  expect(bg.local.css).toMatchSnapshot();
  expect(bg.scroll.css).toMatchSnapshot();
});

test("Background Clip", () => {
  const bg = createUtility("bg")
    .use(backgroundClip(backgroundClipConfig))
    .init();
  expect(bg.clip.border.css).toMatchSnapshot();
  expect(bg.clip.padding.css).toMatchSnapshot();
  expect(bg.clip.content.css).toMatchSnapshot();
  expect(bg.clip.text.css).toMatchSnapshot();
});

test("Background Clip With Different Key", () => {
  const bg = createUtility("bg")
    .use(backgroundClip(backgroundClipConfig, "clipped"))
    .init();
  expect(bg.clipped.border.css).toMatchSnapshot();
  expect(bg.clipped.padding.css).toMatchSnapshot();
  expect(bg.clipped.content.css).toMatchSnapshot();
  expect(bg.clipped.text.css).toMatchSnapshot();
});

test("Background Generic", () => {
  const bg = createUtility("bg")
    .use(backgroundGeneric())
    .init();
  expect(bg["rgb(22, 22, 22)"].css).toMatchSnapshot();
  expect(bg.red.css).toMatchSnapshot();
  expect(bg.aliceBlue.css).toMatchSnapshot();
  expect(bg[0xffffff].css).toMatchSnapshot();
});
