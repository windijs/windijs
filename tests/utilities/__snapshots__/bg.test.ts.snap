// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Attachment 1`] = `
Object {
  "backgroundAttachment": "fixed",
}
`;

exports[`Background Attachment 2`] = `
Object {
  "backgroundAttachment": "local",
}
`;

exports[`Background Attachment 3`] = `
Object {
  "backgroundAttachment": "scroll",
}
`;

exports[`Background Clip 1`] = `
Object {
  "backgroundClip": "border-box",
  "webkitBackgroundClip": "border-box",
}
`;

exports[`Background Clip 2`] = `
Object {
  "backgroundClip": "padding-box",
  "webkitBackgroundClip": "padding-box",
}
`;

exports[`Background Clip 3`] = `
Object {
  "backgroundClip": "content-box",
  "webkitBackgroundClip": "content-box",
}
`;

exports[`Background Clip 4`] = `
Object {
  "backgroundClip": "text",
  "webkitBackgroundClip": "text",
}
`;

exports[`Background Clip With Different Key 1`] = `
Object {
  "backgroundClip": "border-box",
  "webkitBackgroundClip": "border-box",
}
`;

exports[`Background Clip With Different Key 2`] = `
Object {
  "backgroundClip": "padding-box",
  "webkitBackgroundClip": "padding-box",
}
`;

exports[`Background Clip With Different Key 3`] = `
Object {
  "backgroundClip": "content-box",
  "webkitBackgroundClip": "content-box",
}
`;

exports[`Background Clip With Different Key 4`] = `
Object {
  "backgroundClip": "text",
  "webkitBackgroundClip": "text",
}
`;

exports[`Background Color 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color 2`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(255, 255, 255, var(--w-bg-opacity))",
}
`;

exports[`Background Color 3`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(29, 78, 216, var(--w-bg-opacity))",
}
`;

exports[`Background Color 4`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(239, 68, 68, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: black 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: black-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: current 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color With Different Color Value Type: dark 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(28, 28, 30, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: dark-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(28, 28, 30, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsl 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsl-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsla 1`] = `
Object {
  "--w-bg-opacity": "0.3",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hsla-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hsla(360, 100%, 50%, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwb 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwb-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwba 1`] = `
Object {
  "--w-bg-opacity": ".5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: hwba-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "hwb(194 0% 0% / var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: inherit 1`] = `
Object {
  "backgroundColor": "inherit",
}
`;

exports[`Background Color With Different Color Value Type: rgb 1`] = `
Object {
  "--w-bg-opacity": "1",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgb-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgba 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: rgba-opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(22, 13, 14, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Different Color Value Type: transparent 1`] = `
Object {
  "backgroundColor": "transparent",
}
`;

exports[`Background Color With Different Opacity Name 1`] = `
Object {
  "--my-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--my-opacity))",
}
`;

exports[`Background Color With Different Opacity Name 2`] = `
Object {
  "--my-opacity": "0.5",
  "backgroundColor": "rgba(254, 243, 199, var(--my-opacity))",
}
`;

exports[`Background Color With Opacity 1`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(0, 0, 0, var(--w-bg-opacity))",
}
`;

exports[`Background Color With Opacity 2`] = `
Object {
  "--w-bg-opacity": "0.5",
  "backgroundColor": "rgba(254, 243, 199, var(--w-bg-opacity))",
}
`;

exports[`Background Color Without Opacity 1`] = `
Object {
  "backgroundColor": "#000",
}
`;

exports[`Background Color Without Opacity 2`] = `
Object {
  "backgroundColor": "#fef3c7",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: black 1`] = `
Object {
  "backgroundColor": "#000",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: current 1`] = `
Object {
  "backgroundColor": "currentColor",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: dark 1`] = `
Object {
  "backgroundColor": "#1c1c1e",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hsl 1`] = `
Object {
  "backgroundColor": "hsl(360, 100%, 50%)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hsla 1`] = `
Object {
  "backgroundColor": "hsla(360, 100%, 50%, 0.3)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hwb 1`] = `
Object {
  "backgroundColor": "hwb(194 0% 0%)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: hwba 1`] = `
Object {
  "backgroundColor": "hwb(194 0% 0% / .5)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: inherit 1`] = `
Object {
  "backgroundColor": "inherit",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: rgb 1`] = `
Object {
  "backgroundColor": "rgb(22, 13, 14)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: rgba 1`] = `
Object {
  "backgroundColor": "rgba(22, 13, 14, 0.5)",
}
`;

exports[`Background Color Without Opacity With Different Color Value Type: transparent 1`] = `
Object {
  "backgroundColor": "transparent",
}
`;

exports[`Background Generic 1`] = `
Object {
  "backgroundColor": "rgb(22, 22, 22)",
}
`;

exports[`Background Generic 2`] = `
Object {
  "backgroundColor": "red",
}
`;

exports[`Background Generic 3`] = `
Object {
  "backgroundColor": "aliceBlue",
}
`;

exports[`Background Generic 4`] = `
Object {
  "backgroundColor": "#ffffff",
}
`;
